webpack/JobInvocationDetail/JobInvocationHostTable.js

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable camelcase */
+import PropTypes from 'prop-types';
+import React, { useMemo } from 'react';
+import { Tr, Td } from '@patternfly/react-table';
+import { foremanUrl } from 'foremanReact/common/helpers';
+import { useAPI } from 'foremanReact/common/hooks/API/APIHooks';
+import { Table } from 'foremanReact/components/PF4/TableIndexPage/Table/Table';
+import TableIndexPage from 'foremanReact/components/PF4/TableIndexPage/TableIndexPage';
+import { useSetParamsAndApiAndSearch } from 'foremanReact/components/PF4/TableIndexPage/Table/TableIndexHooks';
+import {
+  useBulkSelect,
+  useUrlParams,
+} from 'foremanReact/components/PF4/TableIndexPage/Table/TableHooks';
+import { getControllerSearchProps } from 'foremanReact/constants';
+import JobStatusIcon from '../react_app/components/RecentJobsCard/JobStatusIcon';
+import {
+  columns,
+  JOB_INVOCATION_HOSTS,
+  STATUS,
+} from './JobInvocationConstants';
+
+const JobInvocationHostTable = ({ id }) => {
+  const columnNamesKeys = Object.keys(columns);
+  const apiOptions = { key: JOB_INVOCATION_HOSTS, search: urlSearchQuery };
+  const {
+    search: urlSearchQuery = '',
+    page: urlPage,
+    per_page: urlPerPage,
+  } = useUrlParams();
+  const defaultParams = { search: urlSearchQuery };
+  if (urlPage) defaultParams.page = Number(urlPage);
+  if (urlPerPage) defaultParams.per_page = Number(urlPerPage);
+  const { response, status, setAPIOptions } = useAPI(
+    'get',
+    `/api/job_invocations/${id}/hosts`,
+    {
+      params: { ...defaultParams, key: JOB_INVOCATION_HOSTS },
+    }
+  );
+
+  const { setParamsAndAPI, params } = useSetParamsAndApiAndSearch({
+    defaultParams,
+    apiOptions,
+    setAPIOptions,
+  });
+
+  const combinedResponse = {
+    response: {
+      search: '',
+      can_create: false,
+      results: response?.results || [],
+      total: response?.total || 0,
+      per_page: response?.perPage,
+      page: response?.page,
+      subtotal: response?.subtotal || 0,
+      message: response?.message || 'error',
+    },
+    status,
+    setAPIOptions,
+  };
+
+  const { updateSearchQuery } = useBulkSelect({
+    initialSearchQuery: urlSearchQuery,
+  });
+
+  const controller = 'hosts';
+  const memoDefaultSearchProps = useMemo(
+    () => getControllerSearchProps(controller),
+    [controller]
+  );
+  memoDefaultSearchProps.autocomplete.url = foremanUrl(
+    `/${controller}/auto_complete_search`
+  );
+
+  return (
+    <TableIndexPage
+      apiUrl=""
```

Comment:
Why is it empty?

Reply:
Because it is not used in the `TableIndexPage`, I use the `replacementResponse` instead, however, feel free to correct me.